typings/index.d.ts

```diff
@@ -1913,6 +1913,8 @@ declare module 'discord.js' {
       name: string,
       options: GuildCreateChannelOptions,
     ): Promise<TextChannel | VoiceChannel | CategoryChannel>;
+    public fetch(id: Snowflake, cache?: boolean, force?: boolean): Promise<GuildChannel | null>;
+    public fetch(id?: Snowflake, cache?: boolean, force?: boolean): Promise<Collection<Snowflake, GuildChannel>>;
```

Comment:
```suggestion
    public fetch(): Promise<Collection<Snowflake, GuildChannel>>;
```

Reply:
It may behave incorrectly in such cases, though. Suppose I want to fetch all channels and not cache them, that overwrite will apply there.

Reply:
So you expect the users to call `.fetch(undefined, false)`?

Reply:
That's what you do to not cache the channels. I don't see why shouldn't we support that?

Reply:
Since it's copy paste of that anyway, how does `roles.fetch()` behave there

Reply:
`guild.roles.fetch()` is inherently a hack because it's not possible to fetch an individual role. If this is intended to replicate that same functionality, make sure you refer to #4935 first

Reply:
Yeah, I'm completely aware of that. I PRed #4933 first which already fixes that. Should be fine 👍